### PR TITLE
Reconfigured backend to query SeatGeek API

### PR DESCRIPTION
**Pull request by samifriedrich**
Hi @devinowen, here's a summary of what I've done since we last met. I could use your input on a few questions below.

- I overhauled ShowQuester's backend so that all venue and event data is now pulled from SeatGeek instead of Songkick. Downstream data structures and Spotify pipelines are the same as before, so the frontend will see the same json objects. 
- I restored the frontend to be responsive to the backend by commenting out the hard coded stuff we added (should we keep these bits for testing or just remove entirely before I merge?) 
- I have updated the S3 file of client credentials to include SeatGeek keys. I'll get these to you so that you can add them to your local `secrets.json` file.
- Everything functions beautifully when I run ShowQuester locally, but I anticipate some debugging will be necessary to get everyone to play nice on AWS. Given the frontend is continuously deployed, it seems I'll need to merge this branch into main before I can test on AWS. 
- To reflect the new Elastic Beanstalk environment, I updated the backend URL within the Amplify and Spotify consoles.

Let me know if you can think of anything else I missed that needs to be updated. I'm excited to get the new and improved ShowQuester live!

### CODE REVIEW

File: flaskapp/app/routes.py [hunk 1]
```diff
@@ -116,96 +118,121 @@ def callback():
...
-    """Search for venue and retrieve info from Songkick.
-    To handle duplicate venue names, this function will check that venue's city matches that entered in venues.csv""" 
+def seatgeek_request(endpoint, venue_id = None, params = None):
+    """Format a request to SeatGeek API"""
+    parameters = {'client_id': SEATGEEK_CLIENT_ID,
+                 'client_secret': SEATGEEK_CLIENT_SECRET}
+    if params is not None:
+        for key, value in params.items():
+            parameters[key] = value
+    if venue_id is not None and endpoint == 'venues':
```
Comment (devinowen) [inline flaskapp/app/routes.py:128]: Personally, I would split the different seatgeek API options out into their own smaller functions  (like `get_venue` or `get_events`) and construct the URL and parameter objects in those. They could then of course pass that info to a generic `seatgeek_request` function that handles everything else. That way you don't end up with a long list of conditional logic as your requests to seatgeek grow.

File: flaskapp/app/routes.py [hunk 2]
```diff
@@ -116,96 +118,121 @@ def callback():
...
+        request_base = f'{SEATGEEK_API_BASE}/{endpoint}/{venue_id}'
+    elif venue_id is not None and endpoint == 'events':
+        request_base = f'{SEATGEEK_API_BASE}/{endpoint}'
+        parameters['venue.id'] = venue_id
+    else:
+        request_base = f'{SEATGEEK_API_BASE}/{endpoint}'
+    response = requests.get(request_base, params=parameters)
+    return response
+
+def get_venue_info(venue_name = None, venue_city = "", venue_id = None):
```
Comment (devinowen) [inline flaskapp/app/routes.py:138]: Oh I see, you kinda already have that. Honestly we're probably getting to the point where breaking up this file into a more abstract - but ultimately organized - structure of services, models, etc. is going to be useful. That would help us break up the function above without feeling like we're adding a bunch of new functions to this already long file. I'd say let's chat this over next time we talk.

File: webapp/src/pages/api/createPlaylist.api.ts [hunk 3]
```diff
@@ -6,15 +6,15 @@ const FLASK_APP_URL: string = process.env.FLASK_APP_URL as string;
...
-          ],
-        playlist_tracks: [],
-      }
-    });
+    // return res.status(200).send({
```
Comment (devinowen) [inline webapp/src/pages/api/createPlaylist.api.ts]: If this is working as expected locally I'd say let's just delete this and the comments in the file below.

**Review by devinowen — APPROVED** [review]
Looking good! One comment on code structure, but nothing blocking the merge. It's up to you whether you want to clean up the code now or later.

File: flaskapp/app/routes.py (hunk 2 above)
Comment (samifriedrich) [reply flaskapp/app/routes.py:138]: That would be super helpful! Restructuring is something I've had on my to-do list for a long time.

File: flaskapp/app/routes.py (hunk 1 above)
Comment (samifriedrich) [reply flaskapp/app/routes.py:128]: I could definitely use some help making these functions more interpretable and modular. My logic was to build a separate function that handles building the request itself for any functions that interface with the SeatGeek API. But I realize having a bunch of conditions is not ideal.

**Comment by samifriedrich** [comment]
I merged this for now so I can see how it runs on AWS, but I'll make my next task reorganizing the flask code.

**Pull request merged**